mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+        :type showall: bool
+        :return: projects.
+        :rtype: dict
+        """
+        if not showall:
+            return self._fetch_resource("projects")
+        else:
+            payload = "showall"
+            return self._fetch_resource("projects", parameters=payload)
+
+    def get_my_project(self, name=None, id=None):
+        """
+        Gets a specific MERMAID project. Either name or id required.
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html for more information on project resources.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: project with associated data.
+        :rtype: dict
+        """
+        if id:
+            return self._api_projects_path(id=id)
+        elif name:
+            projects = self.get_projects(showall=True)
+            proj = get_dict_by_keyval("name", name, projects["results"])
+            if proj is None:
+                raise Exception(f"No project with the name:{name}")
+            else:
+                return proj
+
+    def get_project_id(self, name):
+        """
+        Gets an ID for given MERMAID project.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type: str
+        :return: project ID.
+        :rtype: str
+        """
+        project = self.get_my_project(name)
+        # Nested if instead of 'elif name and self.get_my_project(name=name)' to improve performance by reducing
+        # the number of API calls for name or id NoneType check.
+        if project:
+            return project.get("id")
+
+    def get_project_data(self, data, name=None, id=None):
+        """
+        Gets project resource data including resources and observations. Authenticated access to project data
+        depends on the association of a user profile with a project. Unauthenticated access to project data
+        depends on the data sharing policies chosen per survey method for a project.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :param data: project resource, observation or sample unit data.
+        For more info visit https://mermaid-api.readthedocs.io/en/latest/projects.html#project-entity-resources
+        :return: project data.
+        :rtype: dict
+        """
+        # Valid data check.
+        if (
+            data not in self.proj_resources
+            and data not in self.proj_obs
+            and data not in self.samples
+        ):
+            raise Exception(f"Invalid project data:{data}")
+        # Create path from either name or id parameter.
+        return self._api_projects_path(name=name, id=id, resource=data)
+
+    def get_obs_data(self, obs, filter, filter_val=None, name=None, id=None):
```

Comment:
I think this is better `get_observations` to match the convention of the other methods